src/NUnitFramework/framework/TestContext.cs

```diff
@@ -512,13 +512,13 @@ public bool ContainsKey(string key)
             {
                 get
                 {
-                    var list = new List<object>();
-                    foreach(var item in _source[key])
+                    if (_source.TryGet(key, out var values))
```

Comment:
fast case for not finding anything now doesn't allocate a List, using `IEnumerable` API as promised